addon/mixins/model-records/resource-route-mixin.js

```diff
@@ -2,40 +2,26 @@ import Ember from 'ember'
 import EmberDataRouteMixin from 'ember-data-route'
 
 const {
-  computed,
-  getOwner,
   Mixin,
-  String: { singularize }
+  inject: { service },
+  A
 } = Ember
 
-export default Mixin.create(EmberDataRouteMixin, {
-  templateAdminPath: null,
-
-  recordType: computed(function() {
-    return singularize(this.paramsFor('model-records').name)
-  }),
+const relationshipsFor = model => {
+  const relationships = []
 
-  templatePath: computed('recordType', function() {
-    return `${this.get('templateAdminPath')}/${this.get('recordType')}`
-  }),
+  model.get('constructor.relationshipsByName').forEach(relationship =>
+    relationships.push(relationship))
 
-  defaultTemplatePath: computed(function() {
-    return `${this.get('templateAdminPath')}/default`
-  }),
+  return A(relationships)
```

Comment:
This actually gives us the relationship object which has name, type and options (which includes the inverse relationships).  It is true though, seems like there should be an easier way to handle these things but had to move on.